src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/common/PersistenceService.java

```diff
@@ -245,23 +249,96 @@ public DatabaseChangelogLockEntity releaseDatabaseLock(int id) {
+
+        long startTime = System.currentTimeMillis();
+        Specification finalQuery = null;
+        if (CollectionUtils.isNotEmpty(queryRequests.getQueryRequests())) {
+            Specification initialQuery = Specification.where(
+                searchByActorIds(queryRequests.getQueryRequests().get(0).getActorId()))
+                .and(searchByGrantType(queryRequests.getQueryRequests().get(0).getGrantType()))
+                .and(searchByValidDate(queryRequests.getQueryRequests().get(0).getValidAt()))
+                .and(searchByAttributes(queryRequests.getQueryRequests().get(0).getAttributes()))
+                .and(searchByRoleType(queryRequests.getQueryRequests().get(0).getRoleType()))
+                .and(searchByRoleName(queryRequests.getQueryRequests().get(0).getRoleName()))
+                .and(searchByClassification(queryRequests.getQueryRequests().get(0).getClassification()))
+                .and(searchByRoleCategories(queryRequests.getQueryRequests().get(0).getRoleCategory()))
+                .and(searchByAuthorisations(queryRequests.getQueryRequests().get(0).getAuthorisations()))
+                .and(searchByHasAttributes(queryRequests.getQueryRequests().get(0).getHasAttributes()))
+                .and(searchByReadOnly(queryRequests.getQueryRequests().get(0).isReadOnly()));
+
+
+            if (queryRequests.getQueryRequests().size() > 1) {
+                for (int i = 1; i < queryRequests.getQueryRequests().size(); i++) {
+                    finalQuery = initialQuery.or(searchByRoleName(queryRequests.getQueryRequests().get(i)
+                                                                      .getRoleName())
+                           .and(searchByHasAttributes(queryRequests.getQueryRequests().get(i).getHasAttributes()))
+                           .and(searchByAuthorisations(queryRequests.getQueryRequests().get(i).getAuthorisations()))
+                           .and(searchByActorIds(queryRequests.getQueryRequests().get(i).getActorId()))
+                           .and(searchByGrantType(queryRequests.getQueryRequests().get(i).getGrantType()))
+                           .and(searchByValidDate(queryRequests.getQueryRequests().get(i).getValidAt()))
+                           .and(searchByAttributes(queryRequests.getQueryRequests().get(i).getAttributes()))
+                           .and(searchByRoleType(queryRequests.getQueryRequests().get(i).getRoleType()))
+                           .and(searchByClassification(queryRequests.getQueryRequests().get(i).getClassification()))
+                           .and(searchByRoleCategories(queryRequests.getQueryRequests().get(i).getRoleCategory()))
+                           .and(searchByReadOnly(queryRequests.getQueryRequests().get(i).isReadOnly())));
+
+                }
+            } else {
+                finalQuery = initialQuery;
+            }
+
+        }
+
+
+        pageRoleAssignmentEntities = roleAssignmentRepository.findAll(
```

Comment:
We had to add the @RequestScope on this persistence service class..to avoid the cross connection between multiple execution of this singleton class. Ideally we should avoid any data members like "pageRoleAssignmentEntities" on the service class. please find an alternative as we now have time to fix these issues.

Reply:
This class had already @RequestScope and it's not singleton by nature. I am not getting your second point regarding data members.